clang/lib/Sema/SemaExpr.cpp

```diff
@@ -267,6 +267,9 @@ bool Sema::DiagnoseUseOfDecl(NamedDecl *D, ArrayRef<SourceLocation> Locs,
 
     if (getLangOpts().CUDA && !CheckCUDACall(Loc, FD))
       return true;
+
+    if (getLangOpts().SYCLIsDevice)
+      checkSYCLDeviceFunction(Loc, FD);
```

Comment:
Should this return 'true' in the error case like every other situation?

Reply:
IMO, It should return something only if we will validate something inside. Like CUDA does validation of function calls. But for now we don't validate anything inside this function, we only check if we know that this function will be emitted or not.